src/lucky_web/params.cr

```diff
@@ -70,10 +72,35 @@ class LuckyWeb::Params
     @_form_params ||= HTTP::Params.parse(body)
   end
 
+  @_multipart_params : HTTP::Params?
+
+  private def multipart_params
+    @_multipart_params ||= HTTP::Params.parse(multipart_body.to_s)
+  end
+
+  @_multipart_body : String?
+
+  private def multipart_body
+    if @_multipart_body
+      @_multipart_body
+    else
+      HTTP::Multipart.parse(request) do |headers, io|
+        if headers["Content-Type"] == "application/x-www-form-urlencoded"
+          @_multipart_body = io.gets_to_end
+        end
+      end
+      @_multipart_body
+    end
+  end
+
   private def json?
     content_type == "application/json"
   end
 
+  private def multipart?
+    content_type.try { |ct| ct.match /^multipart\// }
```

Comment:
Crystal question - Is there a better way to handle nil-able queries like this?

Reply:
Good question. You can use the block shorthand https://crystal-lang.org/docs/syntax_and_semantics/blocks_and_procs.html to make it more concise: `content_type.try(&.match(multipleart regex))`

You can also call `not_nil!` on it but you should only do that if you expect it to ALWAYS be not nil because it raises at runtime. I consider not_nil to be a code smell and try to avoid it if possible.  I hope that helps :)